packages/dashboard/src/components/dashboard/index.tsx

```diff
@@ -13,33 +13,40 @@ import { DashboardState, SaveableDashboard } from '../../store/state';
 import { PickRequiredOptional, RecursivePartial } from '../../types';
 import { SiteWiseQuery } from '@iot-app-kit/source-iotsitewise';
 import { DashboardMessages, DefaultDashboardMessages } from '../../messages';
+import { ClientContext } from './clientContext';
 
 import '@cloudscape-design/global-styles/index.css';
 import '../../styles/variables.css';
+import { IoTSiteWiseClient } from '@aws-sdk/client-iotsitewise';
 
 export type IotDashboardProps = {
   messageOverrides?: RecursivePartial<DashboardMessages>;
   query?: SiteWiseQuery;
   onSave?: (dashboard: SaveableDashboard) => void;
+  client?: IoTSiteWiseClient;
 } & PickRequiredOptional<DashboardState, 'dashboardConfiguration', 'readOnly' | 'grid'>;
 
-const Dashboard: React.FC<IotDashboardProps> = ({ messageOverrides, query, onSave, ...dashboardState }) => (
-  <Provider store={configureDashboardStore({ ...dashboardState })}>
-    <DndProvider
-      backend={TouchBackend}
-      options={{
-        enableMouseEvents: true,
-        enableKeyboardEvents: true,
-      }}
-    >
-      <InternalDashboard
-        query={query}
-        onSave={onSave}
-        messageOverrides={merge(messageOverrides, DefaultDashboardMessages)}
-      />
-      <WebglContext />
-    </DndProvider>
-  </Provider>
-);
+const Dashboard: React.FC<IotDashboardProps> = ({ messageOverrides, query, onSave, client, ...dashboardState }) => {
+  return (
+    <ClientContext.Provider value={client}>
```

Comment:
currently `describeAsset` uses this object(`packages/dashboard/src/util/IotSitewiseClient.ts`) to get a client instance. Using context is fine. But we need to update `describeAsset` as well.

Reply:
Agreed, but I think we can do this next.